yjit/src/backend/ir.rs

```diff
@@ -859,19 +867,31 @@ pub struct Assembler
 
     /// Names of labels
     pub(super) label_names: Vec<String>,
+
+    /// True if each stack slot of the corresponding index is spilled
+    pub(super) spilled_temps: [bool; 8],
```

Comment:
Does `[bool; 8]` automatically get converted into a `u8` in Rust? If not, might be best to always work with a `u8`. That way it's easier to store in the context without having to translate to/from.

Reply:
It does:

```rs
    println!("size_of u8: {}", std::mem::size_of::<u8>());
    println!("size_of bool: {}", std::mem::size_of::<bool>());
    println!("size_of [u8; 8]: {}", std::mem::size_of::<[u8; 8]>());
    println!("size_of [bool; 8]: {}", std::mem::size_of::<[bool; 8]>());
```

```
size_of u8: 1
size_of bool: 1
size_of [u8; 8]: 8
size_of [bool; 8]: 8
```

Reply:
ah well, no. you're right.

Reply:
It wasn't converted to u8 for the 8 elements. I'll change it to a u8 bitmap.